irods/connection.py

```diff
@@ -80,8 +80,8 @@ def client_signature(self):
         return self._client_signature
 
     def __del__(self):
-        if self.socket and getattr(self,"_disconnected",False):
-            self.disconnect()
+        self.disconnect()
```

Comment:
Moved the conditions to call disconnect() inside the function. Also, added a new criteria for calling disconnect. This makes it safe to call disconnect multiple times on the same connection safely. The first call cleans up the resources and next calls are no-ops.